### PR TITLE
Biels patch add network gamesetup

### LINKED ISSUE #97: make an alternate MagiTech NetworkGameSetup
To help with #48 

We should have a 3rd mgg file in the same directory this is in:

https://github.com/ZetaGlest/zetaglest-data/blob/develop/data/defaultNetworkGameSetup.mgg

The file could be called defaultMagitechNetworkGameSetup.mgg

Anyone wishing to run a headless server could specify that magitech mgg file as the default (cc @biels ;) )

### PR DESCRIPTION
**Pull request by biels**
resolves #97

### CODE REVIEW

File: data/defaultMagitechNetworkGameSetup.mgg [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+FactionTypeNameForIndex8=*Observer*
+FactionPlayerNameForIndex8=???
+FactionPlayerUUIDForIndex8=
+FactionControlForIndex9=5
+ResourceMultiplierIndex9=5
+FactionTeamForIndex9=10
+FactionStartLocationForIndex9=9
+FactionTypeNameForIndex9=*Observer*
+FactionPlayerNameForIndex9=???
+FactionPlayerUUIDForIndex9=
```
Comment (andy5995) [inline data/defaultMagitechNetworkGameSetup.mgg]: @biels Thanks for submitting this. And could you add a short README to this directory to explain how someone wanting to run a headless server would use this file?

**Comment by andy5995** [comment]
@biels commit https://github.com/ZetaGlest/zetaglest-data/pull/129/commits/bb84a37545e7ec5234b3cd7e457964ab3d53babf must be removed from this pull request; it's already included in pr #128 

Remember to switch back to the **develop** branch before working on a 2nd patch. I think what happened is you checked out **biels-patch-add-network-gamesetup** when you were still on the **biels:biels-patch-balance-powergolem** branch. Which means the golem changes that you made carried over and became part of this PR.

I'm not really sure the best way to remove a commit from a branch. You may just want to 

* make a backup of the changes (copy your new game setup file to a location outside the repo)
* checkout develop
* create a new branch
* copy your new mgg file back into the repo, and add the README (which I mentioned in my line comment)
* do a new PR (close this one)

**Comment by mathusummut** [comment]
Should we merge this?

**Comment by biels** [comment]
Removed unrelated commit

**Comment by andy5995** [comment]
>Well I added the file because it the issue told me to do, but I don't know how I would actually use it. Using some existing CLI flag maybe?

@biels , We'll follow up on this tomorrow and try to figure it out.

Let's not merge this yet.

**Comment by andy5995** [comment]
@biels wrote:

>Using some existing CLI flag maybe?

These are the closest we have, but neither will work. Perhaps @mathusummut would be able to use some of the code though to easily add a CLI option.

```
--load-saved-game=x     Loads the last saved game.

                        Where x is an optional name of the saved game file to load.

                        If x is not specified we load the last game that was saved.

--auto-test=x,y,z       Run in auto test mode.

                        Where x is an optional maximum # seconds to play.

                        If x is not specified the default is 1200 seconds (20 minutes).

                        Where y is an optional game settings file to play.

                        If y is not specified (or is empty) then auto test cycles

                        through playing scenarios.

                        Where z is the word 'exit' indicating the game should exit

                        after the game is finished or the time runs out. If z is

                        not specified (or is empty) then auto test continues to cycle.
```

The `load-saved-game` option wants an xml file, not a game settings (.mgg file).

The best way to do this would be to add a 4th option here, to load the desired .mgg file:

```
--headless-server-mode=x,x  

                        Run as a headless server.

                        Where x is an optional comma delimited command list of one or

                            more of the following: 

                        'exit' - which quits the application after a game has no more

                            connected players.

                        'vps' - which does NOT read commands from the local console

                            (for some vps's).

                        'lan' - which does not broadcast the hosting server to the

                            masterserver (for local LAN games).
```

I'll create a ticket for that. We'll leave this PR sit for another day or 2 until we get some feedback from @mathusummut 

I do have one (dirty) short-term solution, but I'll mention it after I hear from Mathu.

**Pull request merged**